### ISSUE TITLE
Low Dice Score for SoftSeg using UNet on T1 as compared to T2

### ISSUE THREAD

**Issue by koushikvikram** [ivadomed/ivadomed#665]
I've trained separate UNet models using SoftSeg for T1 and T2 images with the same hyperparameters. However, the Dice score on both are vastly different. 

Results for T1:

![eval-results](https://user-images.githubusercontent.com/54964630/106895194-ef32f380-6715-11eb-9588-10f0af78d1fc.JPG)

Results for T2:

![eval-results](https://user-images.githubusercontent.com/54964630/106895270-0245c380-6716-11eb-9808-bb22702f1103.JPG)

Config:
```json
{
    "command": "test",
    "gpu": "5",
    "log_directory": "unet_t1_1",
    "model_name": "unet_soft_seg_sc_csf_t1",
    "debugging": true,
    "loader_parameters": {
        "bids_path": "<PATH_HIDDEN>",
        "target_suffix": ["_seg", "_csfseg-manual"],
        "roi_params": {
            "suffix": "_roi",
            "slice_filter_roi": null
        },
        "contrast_params": {
            "training_validation": ["T1w_RPI_r"],
            "testing": ["T1w_RPI_r"],
            "balance": {}
        },
        "slice_filter_params": {
            "filter_empty_mask": false,
            "filter_empty_input": true
            
        },
        "slice_axis": "axial",
        "multichannel": false,
        "soft_gt": true
    },
    "split_dataset": {
        "fname_split": null,
        "random_seed": 6,
        "center_test": [],
        "method": "per_patient",
        "balance": null,
        "train_fraction": 0.7,
        "test_fraction": 0.1
    },
    "training_parameters": {
        "batch_size": 64,
        "loss": {
            "name": "AdapWingLoss"
        },
        "training_time": {
            "num_epochs": 100,
            "early_stopping_patience": 50,
            "early_stopping_epsilon": 0.001
        },
        "scheduler": {
            "initial_lr": 0.001,
            "lr_scheduler": {
                "name": "CosineAnnealingLR",
                "base_lr": 1e-5,
                "max_lr": 1e-2
            }
        },
        "balance_samples": false,
        "mixup_alpha": null,
        "transfer_learning": {
            "retrain_model": null,
            "retrain_fraction": 1.0,
            "reset": true
        }
    },
    "default_model": {
        "name": "Unet",
        "dropout_rate": 0.3,
        "bn_momentum": 0.9,
        "depth": 3,
        "relu": true
    },
    "testing_parameters": {
        "binarize_prediction": -1,
        "uncertainty": {
            "epistemic": false,
            "aleatoric": false,
            "n_it": 10
        }
    },
    "evaluation_parameters": {
        "targetSize": {"unit": "vox", "thr": [20, 100]},
        "removeSmall": {"unit": "vox", "thr": 3},
        "overlap": {"unit": "vox", "thr": 3},
        "uncertainty": {"thr": -1, "suffix": "_unc-vox.nii.gz"}
    },
    "transformation": {
        "Resample":
        {
            "wspace": 0.75,
            "hspace": 0.75,
            "dspace": 1
        },
        "CenterCrop": {
            "size": [128, 128]
        },
        "RandomAffine": {
            "translate": [0.03, 0.03],
            "applied_to": ["im", "gt"],
            "dataset_type": ["training"]
        },
      "NumpyToTensor": {},
      "NormalizeInstance": {"applied_to": ["im"]}
    }
}
```

Below are the predictions for _sub-balgrist04_:

T1 - Volume 0

![vol-0](https://user-images.githubusercontent.com/54964630/106896364-792f8c00-6717-11eb-904b-aabbedda4b77.JPG)

T1 - Volume 1

![vol-1](https://user-images.githubusercontent.com/54964630/106896394-80ef3080-6717-11eb-9cf2-4df31f2afac6.JPG)

T2 - Volume 0

![vol-0](https://user-images.githubusercontent.com/54964630/106896421-89476b80-6717-11eb-876a-aa063c0687ba.JPG)

T2 - Volume 1

![vol-1](https://user-images.githubusercontent.com/54964630/106896439-906e7980-6717-11eb-8067-cb8a3a641367.JPG)

Also, looking at the train (red)/val (blue) loss plot, it appears that the loss for T1 isn't decreasing (model isn't learning).

T1 loss plot:

![train-val-loss](https://user-images.githubusercontent.com/54964630/106896753-ff4bd280-6717-11eb-909a-350b5c9913df.JPG)

T2 loss plot:

![train-val-loss](https://user-images.githubusercontent.com/54964630/106896774-0672e080-6718-11eb-9bfc-0b4c274de047.JPG)

@jcohenadad @andreanne-lemay @leandervaneekelen Any suggestions on where the model might be going wrong?

**Comment by koushikvikram** [comment]
For the purpose of comparison, I've attached results and figures from training UNet on T1 images **without SoftSeg** using the same config.

T1 Results:

![eval-results](https://user-images.githubusercontent.com/54964630/106902216-93209d00-671e-11eb-8ba4-ab0c74cdc05a.JPG)

Predictions for sub-balgrist04:

Volume 0:

![vol-0](https://user-images.githubusercontent.com/54964630/106902302-ab90b780-671e-11eb-82f6-3547de52180b.JPG)

Volume 1:

![vol-1](https://user-images.githubusercontent.com/54964630/106902308-ad5a7b00-671e-11eb-98bb-ed35b4daaccc.JPG)

Loss Plot:

![train-val-loss](https://user-images.githubusercontent.com/54964630/106902378-c7945900-671e-11eb-8fec-90eb3d8b9444.JPG)

**Comment by andreanne-lemay** [comment]
Hi @koushikvikram, thank you for posting this issue. According to your configuration file, it looks like the activation function might not be the right one. For the activation function, you have `relu: True`. Currently, the sigmoid function is used which might cause these results. It should be `final_activation: "relu"`. 

It seems like there is an error in the documentation. PR #612 fixed this issue, but looks like it never got approved and since this PR #650 the config file completely changed. I updated the PR so the documentation should be in order soon.

**Comment by koushikvikram** [comment]
Thank you Andreanne.
